### DISCUSSION TITLE
Enable Code Block Wrapping by default

### DISCUSSION THREAD

**Discussion by jgoux** [facebook/docusaurus#7769] [Q&A]
Hello all,

I'd like to know if it's possible to have code block wrapping activated for all code blocks automatically?

**Comment by slorber** [comment]
Unfortunately we don't provide any option for that atm

You can only achieve it by swizzle eject `@theme/CodeBlock/Content/String` => not ideal

What's the motivation to do so? can you show a site where not wrapping by default is a problem?

**Reply by jgoux** [reply]
I'm writing the doc for an upcoming feature and there is a lot of CLI commands with pretty long lines, the reader will most likely click on every wrap button: https://deploy-preview-50--snaplet-docs.netlify.app/references/instant-database

**Reply by slorber** [reply]
I see thanks

Not sure it would always be better, but we should give you the opportunity to configure it and figure out for yourself.

Should the button remains present to disable line breaks?

Should you be able to configure it on a code-block basis with code block metastring?

I mean:

````bash

```bash wordWrap=true
$ snaplet database --help

snaplet database [action]

Commands:
  snaplet database create [database-name]  create an instant database from a snapshot         [aliases: c]
  snaplet database delete [database-name]  delete an instant database                         [aliases: d]
  snaplet database list                    list instant databases                             [aliases: ls]
  snaplet database url [database-name]     show an instant database url                       [aliases: u]
  snaplet database cache [snapshot]        cache a snapshot into the instant database server  [aliases: ca]
```

````

**Reply by jgoux** [reply]
Yep, I think the button staying present is good, I want to be able to define the initial state as it's the most sensible default in my context.

Code block meta string would be great, I like this syntax a lot. 👍

**Reply by zakmandhro** [reply]
We need this too. Any update on this feature of setting wordwrap to be on by default?

**Reply by slorber** [reply]
Let's track this in https://github.com/facebook/docusaurus/issues/7875 @jgoux @zakmandhro

**Reply by daninus14** [reply]
would it be possible to at least pass in an option like we have for `title="Some Title"` to also have `wrap=True`?
